File: BaseWork/app/src/main/java/com/zihuan/app/activity/MainActivity.java
package com.zihuan.app.activity;


import android.graphics.Color;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;

import com.zihuan.app.R;
import com.zihuan.app.adapter.ViewPAdapter;
import com.zihuan.app.fragment.Fm_1;
import com.zihuan.app.fragment.Fm_2;
import com.zihuan.app.fragment.Fm_3;
import com.zihuan.app.fragment.Fm_4;
import com.zihuan.app.model.Test;
import com.zihuan.app.u.Logger;
import com.zihuan.app.view.NoScrollViewPager;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import me.majiajie.pagerbottomtabstrip.NavigationController;
import me.majiajie.pagerbottomtabstrip.PageNavigationView;
import me.majiajie.pagerbottomtabstrip.item.BaseTabItem;
import me.majiajie.pagerbottomtabstrip.item.NormalItemView;

public class MainActivity extends BaseActivity {

    @BindView(R.id.vp_main)
    NoScrollViewPager m_vpMain;
    ViewPAdapter mVPAdapter;
    @BindView(R.id.page_bar)
    PageNavigationView pageBar;

    private int anInt;
    NavigationController mNavigationController;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void initView() {
        int color = getResources().getColor(R.color.colorPrimary);
        int def = getResources().getColor(R.color.color_8888);

        mNavigationController = pageBar.material()
                .addItem(R.mipmap.ic_launcher, "首页", color)
                .addItem(R.mipmap.ic_launcher, "第二页", color)
                .addItem(R.mipmap.ic_launcher, "第三页", color)
                .addItem(R.mipmap.ic_launcher, "我的", color)
                .setDefaultColor(def)
                .build();
//        mNavigationController = pageBar.custom()
//                .addItem(newItem(R.mipmap.ic_launcher, "首页", color))
//                .addItem(newItem(R.mipmap.ic_launcher, "第二页", color))
//                .addItem(newItem(R.mipmap.ic_launcher, "第三页", color))
//                .addItem(newItem(R.mipmap.ic_launcher, "我的", color))
////                .setDefaultColor(def)
//                .build();
        List<Fragment> fragments = new ArrayList<>();
        Fm_1 home = new Fm_1();
        Fm_2 page1 = new Fm_2();
        Fm_3 page2 = new Fm_3();
        Fm_4 page3 = new Fm_4();
        fragments.add(home);
        fragments.add(page1);
        fragments.add(page2);
        fragments.add(page3);
        mVPAdapter = new ViewPAdapter(getSupportFragmentManager(), fragments);
        m_vpMain.setAdapter(mVPAdapter);
        mNavigationController.setupWithViewPager(m_vpMain);
//        预加载3个页面
        m_vpMain.setOffscreenPageLimit(3);
        m_vpMain.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                anInt = position;
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        Test test = Test.builder().name("蛤蛤").build();
        Logger.tag(test.toString());
    }

    @Override
    public void initData() {

    }


    //创建一个Item
    private BaseTabItem newItem(int drawable, String text, int checkedDrawable) {
        NormalItemView normalItemView = new NormalItemView(this);
        normalItemView.initialize(drawable, checkedDrawable, text);
        normalItemView.setTextDefaultColor(Color.GRAY);
        normalItemView.setTextCheckedColor(0xFF009688);
        return normalItemView;
    }


}

File: BaseWork/app/src/main/java/com/zihuan/app/ui/activity/MainActivity.kt
package com.zihuan.app.ui.activity


import android.view.KeyEvent
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.google.android.material.bottomnavigation.LabelVisibilityMode
import com.jaeger.library.StatusBarUtil
import com.zihuan.app.R
import com.zihuan.app.adapter.ViewPAdapter
import com.zihuan.app.base.BaseActivity
import com.zihuan.app.model.EventData
import com.zihuan.app.ui.fragment.Fm_2
import kotlinx.android.synthetic.main.activity_main.*
import org.greenrobot.eventbus.Subscribe
import org.jetbrains.anko.toast

class MainActivity : BaseActivity() {

    override fun getLayoutId() = R.layout.activity_main

    override fun initView() {
        StatusBarUtil.setTranslucentForImageViewInFragment(this@MainActivity, null)
        val fragments = ArrayList<Fragment>()
        fragments.add(Fm_2())
        fragments.add(Fm_2())
        fragments.add(Fm_2())
        fragments.add(Fm_2())
        vp_main.adapter = ViewPAdapter(supportFragmentManager, fragments)
        //        预加载3个页面
        vp_main.offscreenPageLimit = 5
        bnve.setupWithViewPager(vp_main)
        bnve.labelVisibilityMode = LabelVisibilityMode.LABEL_VISIBILITY_SELECTED
        bnve.isItemHorizontalTranslationEnabled = false
        vp_main.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrollStateChanged(state: Int) {}
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                when (position) {
                }
            }
        })
        enableEventBus()
    }


    override fun initData() {
    }

    @Subscribe
    fun onEventMainThread(ed: EventData) {
        when (ed.what) {
        }
    }


    private var exitTime: Long = 0

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK && event.action == KeyEvent.ACTION_DOWN) {
            if (System.currentTimeMillis() - exitTime > 2000) {
                exitTime = System.currentTimeMillis()
                toast("再按一次退出")
            } else {
                finish()
            }
            return true
        }
        return super.onKeyDown(keyCode, event)
    }


}

